native/src/main/java/io/ballerina/stdlib/observe/nativeimpl/OpenTracerBallerinaWrapper.java

```diff
@@ -184,9 +184,10 @@ public Object addTag(Environment env, String tagKey, String tagValue, long spanI
         if (spanId == -1) {
             ObserverContext observer = ObserveUtils.getObserverContextOfCurrentFrame(env);
             if (observer == null) {
-                return ErrorCreator.createError(
-                        StringUtils.fromString(
-                                ("Span already finished. Can not add tag {" + tagKey + ":" + tagValue + "}")));
+                return null;
+//                return ErrorCreator.createError(
```

Comment:
Shall we removed the code. Add comment as well. Observer context will be null if function is executed without entry point like main or resource function ex.  initialising phase.

Reply:
@warunalakshitha addressed the review suggestions